docker/resource_docker_service_funcs.go

```diff
@@ -1302,6 +1287,29 @@ func fromRegistryAuth(image string, authConfigs map[string]types.AuthConfig) typ
 	return types.AuthConfig{}
 }
 
+// retrieveAndMarshalAuth retrieves and marshals the service registry auth
+func retrieveAndMarshalAuth(d *schema.ResourceData, meta interface{}, stageType string) []byte {
+	auth := types.AuthConfig{}
+	if v, ok := d.GetOk("auth"); ok {
+		auth = authToServiceAuth(v.(map[string]interface{}))
+	} else {
+		authConfigs := meta.(*ProviderConfig).AuthConfigs.Configs
+		if len(authConfigs) == 0 {
+			log.Printf("[DEBUG] AuthConfigs empty on %s. Wait 3s and try again", stageType)
+			// sometimes the dockerconfig is read succesfully from disk but the
```

Comment:
Just for my comprehension, where/how is this config created ?

Reply:
It's created here when the provider is initialized: https://github.com/terraform-providers/terraform-provider-docker/blob/master/docker/provider.go#L174